### PR TITLE
Feature/alcs 1858 update condition pills

### PR DESCRIPTION
**Pull request by fbarreta**
PG functions to determine condition statuses, and UI implementation

### CODE REVIEW

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts [hunk 1]
```diff
@@ -83,10 +82,6 @@ export class DecisionConditionComponent implements OnInit, OnChanges {
       description: this.data.description ?? null,
     });
 
-    if (this.showSingleDateField && this.dates.length > 0 && this.dates[0].date) {
```
Comment (trslater) [inline alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts:86]: Why was this removed?

File: alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts [hunk 2]
```diff
@@ -83,10 +82,6 @@ export class DecisionConditionComponent implements OnInit, OnChanges {
       description: this.data.description ?? null,
     });
 
-    if (this.showSingleDateField && this.dates.length > 0 && this.dates[0].date) {
```
Comment (trslater) [inline alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts:86]: Why was this removed?

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts (hunk 1 above)
Comment (fbarreta) [reply alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts:86]: Because singleDate field was removed from ApplicationDecisionConditionDto and replaced by dates.

File: alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts (hunk 2 above)
Comment (fbarreta) [reply alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts:86]: Same as above:
Because singleDate field was removed from ApplicationDecisionConditionDto and replaced by dates.

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts [hunk 3]
```diff
@@ -56,10 +56,10 @@ export class DecisionConditionComponent implements OnInit, OnChanges {
 
   form = new FormGroup({
     securityAmount: this.securityAmount,
-    singleDate: this.singleDate,
```
Comment (trslater) [inline alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts:59]: We also need this field.

File: alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts [hunk 4]
```diff
@@ -56,10 +56,10 @@ export class DecisionConditionComponent implements OnInit, OnChanges {
 
   form = new FormGroup({
     securityAmount: this.securityAmount,
-    singleDate: this.singleDate,
```
Comment (trslater) [inline alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts:59]: And this one

File: alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts (hunk 3 above)
Comment (fbarreta) [reply alcs-frontend/src/app/features/application/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts:59]: It's added at the end of the group:
form = new FormGroup({
    securityAmount: this.securityAmount,
    administrativeFee: this.administrativeFee,
    description: this.description,
    componentsToCondition: this.componentsToCondition,
    singleDate: this.singleDate,
  });

File: alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts (hunk 4 above)
Comment (fbarreta) [reply alcs-frontend/src/app/features/notice-of-intent/decision/decision-v2/decision-input/decision-conditions/decision-condition/decision-condition.component.ts:59]: Same here:
form = new FormGroup({
    securityAmount: this.securityAmount,
    administrativeFee: this.administrativeFee,
    description: this.description,
    componentsToCondition: this.componentsToCondition,
    singleDate: this.singleDate,
  });

**Review by Abradat — APPROVED** [review]
Conversations seem to be fixed. LGTM
